modules/fluid_properties/src/fluidproperties/IdealGasFluidProperties.C

```diff
@@ -313,7 +315,7 @@ IdealGasFluidProperties::s_from_p_T(Real p, Real T, Real & s, Real & ds_dp, Real
 Real
 IdealGasFluidProperties::s_from_h_p(Real h, Real p) const
 {
-  const Real aux = p * std::pow(h / (_gamma * _cv), -_gamma / (_gamma - 1));
+  const Real aux = p * std::pow((h - _e_ref) / (_gamma * _cv), -_gamma / (_gamma - 1));
```

Comment:
```suggestion
  // h=h_ref because p_ref = 0
  const Real aux = p * std::pow((h - _e_ref) / (_gamma * _cv), -_gamma / (_gamma - 1));
```

Reply:
I don't understand this comment. Not sure what p_ref is

Reply:
h = e + p v
so for h_ref to be equal to e_ref, you mean p_ref = 0

Reply:
I'm not defining any `h_ref`, just using
h = e + p v
h = e_ref + cv T + p v
so to preserve the original quantity `cv T  + p v`, it's `h - e_ref`

Reply:
ok